tests/conftest.py

```diff
@@ -53,34 +53,55 @@ def pytest_addoption(parser):
 
 
 class AsicDbValidator(DVSDatabase):
-    def __init__(self, db_id: str, connector: str):
+    def __init__(self, db_id: int, connector: str):
         DVSDatabase.__init__(self, db_id, connector)
+        self._wait_for_asic_db_to_initialize()
+        self._populate_default_asic_db_values()
+        self._generate_oid_to_interface_mapping()
 
-        # Get default .1Q Vlan ID
-        self.default_vlan_id = self.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_VLAN", 1)[0]
+    def _wait_for_asic_db_to_initialize(self) -> None:
+        """Wait up to 30 seconds for the default fields to appear in ASIC DB."""
+        def _verify_db_contents():
+            # We expect only the default VLAN
+            if len(self.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_VLAN")) != 1:
```

Comment:
this might be tricky if we load the T1 topology, do we still expect vlan?

Reply:
This check has been here for a while so I'm inclined to leave it as is for this PR, but I can open another one to make this more flexible w.r.t. the topology.

This check as a whole isn't very robust, it would be good if we could pass in an expected topology for the ASIC DB validator to refer to rather than hard-coding it in `conftest.py`.